app/models/marc_batch.rb

```diff
@@ -0,0 +1,38 @@
+class MarcBatch
+  def initialize(theses, marc_filename, zip_filename)
+    @theses = theses
+    @marc_filename = marc_filename
+    @zip_filename = zip_filename
+  end
+
+  def build
+    marc_file = Tempfile.new(@marc_filename)
+    zip_file = Tempfile.new(@zip_filename, binmode: true)
+    create_marc_file(marc_file)
+    zip_marc_file(zip_file, marc_file)
+  end
+
+  private
+
+  def create_marc_file(marc_file)
+    writer = MARC::XMLWriter.new(marc_file.path)
+    @theses.each do |thesis|
+      record = Marc.new(thesis)
+      writer.write(record.record)
+    end
+    writer.close
+  end
+
+  def zip_marc_file(zip_file, marc_file)
+    Zip::File.open(zip_file.path, Zip::File::CREATE) do |zip|
+      zip.add(@marc_filename, marc_file.path)
+    end
+    zip_file.close
+    marc_file.unlink
+
+    # We are reading the file here so we can unlink the tempfile and pass the binary data directly to the mailer.
```

Comment:
Yeah, that's a good point. I'll take a closer look at how ruby handles tempfiles before we move forward with this approach.

Reply:
@JPrevost It occurred to me that we can just close/unlink the tempfile in an `ensure block` in the job. However, ActiveJob doesn't accept tempfiles as arguments, meaning we wouldn't be able to use `deliver_later` to send the email. We could use `deliver_now`, since that sends the email immediately without enqueueing it. Apparently we already use `deliver_now` to send the publication results email, though I'm not sure why: https://github.com/MITLibraries/thing/blob/main/app/jobs/dspace_publication_results_job.rb#L23

Thoughts on this? It seems to me that any risk in using `deliver_now` is outweighed by the benefits of not reading the zip file into memory.